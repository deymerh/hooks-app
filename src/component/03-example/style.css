body {
  padding: 50px;
}
.right {
  text-align: right;
}
