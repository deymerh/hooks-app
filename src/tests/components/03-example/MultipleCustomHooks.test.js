import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../component/03-example/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';
//NO me inporta la implementacion solo me interesan los resultados
//por eso usamos mosk
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Pruebas en el componentes <MultipleCustomHooks />', () => {

  beforeEach(() => {
    useCounter.mockReturnValue({
      counter: 1,
      increment: () => { }
    });
  });

  test('Debe mostrar el componente correctamente', () => {
    useFetch.mockReturnValue({
      loading: true,
      error: null,
      data: null
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar la informacion de manera correacta', () => {
    useFetch.mockReturnValue({
      loading: false,
      error: null,
      data: [{
        author: 'Deymer',
        quote: 'Hola mundo'
      }]
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper.find('.alert').exists()).toBe(false);
    expect(wrapper.find('.mb-0').text().trim()).toBe('Hola mundo');
    expect(wrapper.find('footer').text().trim()).toBe('Deymer');
    // console.log(wrapper.html());
  });


});
