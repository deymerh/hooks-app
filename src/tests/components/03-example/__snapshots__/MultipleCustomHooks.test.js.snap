// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componentes <MultipleCustomHooks /> Debe mostrar el componente correctamente 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <p
    className="alert alert-info text-center"
  >
    Loading...
  </p>
</div>
`;
